tests/test_globaliostream.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+    tests/test_globaliostream.cpp -- Usage of global scoped_output_redirect
+
+    Copyright (c) 2017 Henry F. Schreiner
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+
+#include <pybind11/iostream.h>
+#include "pybind11_tests.h"
+#include <iostream>
+
+TEST_SUBMODULE(globaliostream, m) {
+    // test_evals
+    m.def("redirect_output", []() {
+            return py::capsule(new py::scoped_output_redirect(
```

Comment:
I see that this is going for an RAII-style guard, but those should really be implemented using context managers in Python. Something like the following (not actually tested):

```c++
struct RedirectOutput {
    std::unique_ptr<py:: scoped_output_redirect> redirect;
}

py::class_<RedirectOutput>(m, "redirect_output")
    .def("__enter__", [](RedirectOutput &self) {
        self.redirect.reset(new py::scoped_output_redirect(...))
    })
    .def("__exit__", [](RedirectOutput &self, py::args) {
        self.redirect.reset();
    });
```

Note: it's convention for Python context managers to have lowercase names.

```python
with m.redirect_output():
    # redirected here
    with capture:
        # can be nested
```